data/pigui/modules/system-view-ui.lua

```diff
@@ -0,0 +1,523 @@
+		Engine.SystemMapSetColor(key, svColor[key])
+	end
+end
+
+local function showDvLine(leftIcon, resetIcon, rightIcon, key, Formatter, leftTooltip, resetTooltip, rightTooltip)
+	local wheel = function()
+		if ui.isItemHovered() then
+			local w = ui.getMouseWheel()
+			if w ~= 0 then
+				Engine.TransferPlannerAdd(key, w * 10)
+			end
+		end
+	end
+	local press = ui.coloredSelectedIconButton(leftIcon, mainButtonSize, false, mainButtonFramePadding, svColor.BUTTON_BACK, svColor.BUTTON_INK, leftTooltip)
+	if press or (key ~= "factor" and ui.isItemActive()) then
+		Engine.TransferPlannerAdd(key, -10)
+	end
+	wheel()
+	ui.sameLine()
+	if ui.coloredSelectedIconButton(resetIcon, mainButtonSize, false, mainButtonFramePadding, svColor.BUTTON_BACK, svColor.BUTTON_INK, resetTooltip) then
+		Engine.TransferPlannerReset(key)
+	end
+	wheel()
+	ui.sameLine()
+	press = ui.coloredSelectedIconButton(rightIcon, mainButtonSize, false, mainButtonFramePadding, svColor.BUTTON_BACK, svColor.BUTTON_INK, rightTooltip)
+	if press or (key ~= "factor" and ui.isItemActive()) then
+		Engine.TransferPlannerAdd(key, 10)
+	end
+	wheel()
+	ui.sameLine()
+	local speed, speed_unit = Formatter(Engine.TransferPlannerGet(key))
+	ui.text(speed .. " " .. speed_unit)
+	return 0
+end
+
+local time_selected_button_icon = icons.time_center
+
+local function timeButton(icon, tooltip, factor)
+	if ui.coloredSelectedIconButton(icon, mainButtonSize, false, mainButtonFramePadding, svColor.BUTTON_BACK, svColor.BUTTON_INK, tooltip) then
+		time_selected_button_icon = icon
+	end
+	local active = ui.isItemActive()
+	if active then
+		Engine.SystemMapAccelerateTime(factor)
+	end
+	ui.sameLine()
+	return active
+end
+
+local function loop3items(a, b, c) return { [a] = b, [b] = c, [c] = a } end
+
+local ship_drawing = "SHIPS_OFF"
+local show_lagrange = "LAG_OFF"
+local show_grid = "GRID_OFF"
+local nextShipDrawings = loop3items("SHIPS_OFF", "SHIPS_ON", "SHIPS_ORBITS")
+local nextShowLagrange = loop3items("LAG_OFF", "LAG_ICON", "LAG_ICONTEXT")
+local nextShowGrid = loop3items("GRID_OFF", "GRID_ON", "GRID_AND_LEGS")
+
+local function calcWindowWidth(buttons)
+	return (mainButtonSize.y + mainButtonFramePadding * 2) * buttons
+	+ itemSpacing.x * (buttons + 1)
+end
+
+local function calcWindowHeight(buttons, separators, texts)
+	return
+	(mainButtonSize.y + mainButtonFramePadding * 2) * buttons
+	+ separators * itemSpacing.y
+	+ texts * ui.fonts.pionillium.medium.size
+	+ (buttons + texts - 1) * itemSpacing.y
+	+ itemSpacing.x * 2
+end
+
+local orbitPlannerWindowPos = Vector2(ui.screenWidth - calcWindowWidth(7), ui.screenHeight - calcWindowHeight(7, 3, 2))
+
+local function showOrbitPlannerWindow()
+	ui.setNextWindowPos(orbitPlannerWindowPos, "Always")
+	ui.withStyleColors({["WindowBg"] = svColor.WINDOW_BACK}, function()
+			ui.window("OrbitPlannerWindow", {"NoTitleBar", "NoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings", "AlwaysAutoResize"},
+								function()
```

Comment:
Looks like there's some indentation weirdness going on here. We use a single level of tabbed (not spaced!) indentation in our lua scripts for each scope change; right here, you have what looks like two tabs of indentation on the previous line followed by an extra three tabs on this line.

I'd appreciate it if you could normalize your indentation. :smile:

Reply:
Sorry, I forgot to indent the code. Fixed 704ec4a66cb4f3467d3897b703a73a8687356223